scripts/runtests.py

```diff
@@ -0,0 +1,128 @@
+import json
+import re
+import subprocess
+import sys
+from pathlib import Path
+from typing import Optional
+
+try:
+    from termcolor import colored
+except ImportError:
+
+    def colored(text: str, color: str = "") -> str:  # type: ignore[misc]
+        return text
+
+
+# Popular Flake8 extensions not used in Typeshed that cause errors
+_STRICTER_CONFIG_FILE = "pyrightconfig.stricter.json"
+_SUCCESS = colored("Success", "green")
+_SKIPPED = colored("Skipped", "yellow")
+_FAILED = colored("Failed", "red")
+
+
+def _parse_jsonc(json_text: str) -> str:
+    # strip comments from the file
+    lines = [line for line in json_text.split("\n") if not line.strip().startswith("//")]
+    # strip trailing commas from the file
+    valid_json = re.sub(r",(\s*?[\}\]])", r"\1", "\n".join(lines))
+    return valid_json
+
+
+def _get_strict_params(stub_path: str) -> list[str]:
+    with open(_STRICTER_CONFIG_FILE) as file:
+        data = json.loads(_parse_jsonc(file.read()))
+    if stub_path in data["exclude"]:
+        return []
+    return ["-p", _STRICTER_CONFIG_FILE]
+
+
+if __name__ == "__main__":
+    try:
+        path = sys.argv[1]
+    except IndexError:
+        print("Missing path argument in format <folder>/<stub>", file=sys.stderr)
+        sys.exit(1)
+    path_tokens = Path(path).parts
+    assert len(path_tokens) == 2, "Path argument should be in format <folder>/<stub>"
+    folder, stub = path_tokens
+    assert folder in {"stdlib", "stubs"}, "Only the 'stdlib' and 'stubs' folders are supported"
+    stubtest_result: Optional[subprocess.CompletedProcess[bytes]] = None
+    pytype_result: Optional[subprocess.CompletedProcess[bytes]] = None
+
+    # Run formatters first. Order matters.
+    subprocess.run(["python3", "-m", "pycln", path, "--all"])
+    subprocess.run(["python3", "-m", "isort", path])
+    subprocess.run(["python3", "-m", "black", path])
+
+    flake8_result = subprocess.run(["python3", "-m", "flake8", path])
+
+    check_consistent_result = subprocess.run(["python3", "tests/check_consistent.py"])
+    check_new_syntax_result = subprocess.run(["python3", "tests/check_new_syntax.py"])
+
+    pyright_result = subprocess.run(["python3", "tests/pyright_test.py", path] + _get_strict_params(path), stderr=subprocess.PIPE)
```

Comment:
I just tried running the script locally in my virtual environment, and this was the output I got:

```console
(.venv) C:\Users\alexw\coding\typeshed>python scripts/runtests.py stubs/sqlalchemy
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.

Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.
Skipping stubtest since mypy failed.
Skipping pytype on Windows. You can run the test with WSL.
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.

One or more tests failed. See above for details.
flake8: Failed
Check consistent: Failed
Check new syntax: Failed
pyright: Failed
mypy: Failed
stubtest: Skipped
pytype: Skipped
Regression test: Failed
```

You need to use `sys.executable` instead of hardcoding `python3` in these `subprocess.run()` calls:

```suggestion
    subprocess.run([sys.executable, "-m", "pycln", path, "--all"])
    subprocess.run([sys.executable, "-m", "isort", path])
    subprocess.run([sys.executable, "-m", "black", path])

    flake8_result = subprocess.run([sys.executable, "-m", "flake8", path])

    check_consistent_result = subprocess.run([sys.executable, "tests/check_consistent.py"])
    check_new_syntax_result = subprocess.run([sys.executable, "tests/check_new_syntax.py"])

    pyright_result = subprocess.run([sys.executable, "tests/pyright_test.py", path] + _get_strict_params(path), stderr=subprocess.PIPE)
```

Reply:
I forgot Microsoft did that weird thing where python3 is just an executable that references/launches the microsoft store... (Until you install python from the store).

Wouldn't that happen in other scripts as well? It doesn't happen for me because I overwrote that shortcut.

Reply:
> Wouldn't that happen in other scripts as well?

Yup, that's why the other scripts in typeshed mostly use `sys.executable` in `subprocess.run()` calls :)

I guess our docs aren't ideal for Windows users, given that they recommend running the scripts with `python3`. But docs get really long and convoluted if you spell out the platform-specific ways of doing each and every thing ://